test/kube_checks.go

```diff
@@ -88,13 +101,20 @@ func WaitForPodState(ctx context.Context, client kubernetes.Interface, inState f
 	span := logging.GetEmitableSpan(ctx, "WaitForPodState/"+name)
 	defer span.End()
 
-	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
-		p, err := p.Get(ctx, name, metav1.GetOptions{})
+	var lastState *corev1.Pod
+	waitErr := wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		var err error
+		lastState, err = p.Get(ctx, name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
-		return inState(p)
+		return inState(lastState)
 	})
+
+	if waitErr != nil {
+		return fmt.Errorf("pod %q is not in desired state, got: %+v: %w", name, spew.Sprint(lastState), waitErr)
```

Comment:
```suggestion
		return fmt.Errorf("pod %q is not in desired state, got: %s: %w", name, spew.Sprint(lastState), waitErr)
```

just a nit, but for completeness sake.